### PR TITLE
 CI new numpy nightly failing some tests

### LINKED ISSUE #50267: CI new numpy nightly failing some tests
```
2022-12-15T09:04:34.1328103Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex] - TypeError: a must be an array of real numbers
2022-12-15T09:04:34.1328576Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex64] - TypeError: a must be an array of real numbers
2022-12-15T09:04:34.1329046Z FAILED pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex128] - TypeError: a must be an array of real numbers
```

Can reproduce this locally with:
```
pip uninstall numpy -y
pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
pytest pandas/tests/series/methods/test_describe.py::TestSeriesDescribe::test_numeric_result_dtype[complex]
```

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #50267 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/methods/test_describe.py [hunk 1]
```diff
@@ -163,6 +166,12 @@ def test_numeric_result_dtype(self, any_numeric_dtype):
             dtype = "complex128" if is_complex_dtype(any_numeric_dtype) else None
 
         ser = Series([0, 1], dtype=any_numeric_dtype)
+        if dtype == "complex128" and is_numpy_dev:
+            with pytest.raises(
+                TypeError, match=r"^a must be an array of real numbers$"
```
Comment (phofl) [inline pandas/tests/series/methods/test_describe.py:171]: Do you know why this isn't allowed anymore?

Comment (MarcoGorelli) [reply pandas/tests/series/methods/test_describe.py:171]: I think it should always have raised - complex numbers aren't ordered, I don't see how it used to work

Comment (MarcoGorelli) [reply pandas/tests/series/methods/test_describe.py:171]: indeed https://github.com/numpy/numpy/issues/22652

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @MarcoGorelli

**Pull request merged**